### PR TITLE
changes to _helpers.tpl in order to fix error

### PR DESCRIPTION
**Pull request by bn-openx**
Got these issues when installing gremlin via helm:

`helm install gremlin/gremlin \ 
\--namespace gremlin \
\--name gremlin \
\--set gremlin.teamID=$GREMLIN_TEAM_ID \
\--set gremlin.clusterID=$GREMLIN_CLUSTER_ID
Error: parse error in "gremlin/templates/_helpers.tpl": template: gremlin/templates/_helpers.tpl:41: unexpected "=" in operand`

`helm install gremlin/gremlin \ 
\--namespace gremlin \
\--name gremlin \
\--set gremlin.teamID=$GREMLIN_TEAM_ID \
\--set gremlin.clusterID=$GREMLIN_CLUSTER_ID
Error: parse error in "gremlin/templates/_helpers.tpl": template: gremlin/templates/_helpers.tpl:43: unexpected "=" in operand`

`helm install ./ \                                                                            
\--namespace gremlin \
\--name gremlin \
\--set gremlin.teamID=$GREMLIN_TEAM_ID \
\--set gremlin.clusterID=$GREMLIN_CLUSTER_ID  
Error: release gremlin failed: DaemonSet.apps "gremlin" is invalid: [spec.template.spec.volumes[4].secret.secretName: Required value, spec.template.spec.containers[0].volumeMounts[4].name: Not found: "gremlin-cert"]`

The PR I am submitting are the changes I made to fix the helm chart and get everything to work. Please let me know if there's anything else you need from me. I would also like to mention that doing a helm lint does not catch this.

### CODE REVIEW

**Comment by gremlin-thor** [comment]
Thanks for the PR! I have a couple of questions so var as I'm working through this:

- What version of helm are you using? Can you post the output of `helm version`?
- Does `helm template` work for you?

**Comment by bn-openx** [comment]
> Thanks for the PR! I have a couple of questions so var as I'm working through this:
> 
> * What version of helm are you using? Can you post the output of `helm version`?

`helm-2.11.0 version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
`
> * Does `helm template` work for you?

`helm template ./` from my local gremlin repo does work correctly (can provide output but it's quite long).

File: gremlin/templates/_helpers.tpl [hunk 1]
```diff
@@ -36,11 +36,11 @@ Because we've evolved the recommended way to pass the secret name over time, we
 In later versions of this chart, we will remove the use of `.Values.gremlin.client.secretName` and the fallback value of `gremlin-team-cert`
 */}}
 {{- define "gremlin.secretName" -}}
-{{- $defaultName := "" -}}
+{{- $defaultName := "gremlin-team-cert" -}}
```
Comment (philgebhardt) [inline gremlin/templates/_helpers.tpl:39]:
```suggestion
```

File: gremlin/templates/_helpers.tpl [hunk 2]
```diff
@@ -36,11 +36,11 @@ Because we've evolved the recommended way to pass the secret name over time, we
...
 {{- if .Values.gremlin.secret.managed -}}
-{{- $defaultName = "gremlin-secret" -}}
+{{- $defaultName := "gremlin-secret" -}}
```
Comment (philgebhardt) [inline gremlin/templates/_helpers.tpl:41]:
```suggestion
{{- default .Values.gremlin.client.secretName .Values.gremlin.secret.name | default "gremlin-secret" -}}
```

File: gremlin/templates/_helpers.tpl [hunk 3]
```diff
@@ -36,11 +36,11 @@ Because we've evolved the recommended way to pass the secret name over time, we
...
 {{- else -}}
-{{- $defaultName = "gremlin-team-cert" -}}
+{{- $defaultName := "gremlin-team-cert" -}}
```
Comment (philgebhardt) [inline gremlin/templates/_helpers.tpl:43]:
```suggestion
{{- default .Values.gremlin.client.secretName .Values.gremlin.secret.name | default "gremlin-team-cert" -}}
```

File: gremlin/templates/_helpers.tpl [hunk 4]
```diff
@@ -36,11 +36,11 @@ Because we've evolved the recommended way to pass the secret name over time, we
...
 {{- end -}}
 {{- default .Values.gremlin.client.secretName .Values.gremlin.secret.name | default $defaultName -}}
```
Comment (philgebhardt) [inline gremlin/templates/_helpers.tpl:45]:
```suggestion
```

**Review by philgebhardt — CHANGES_REQUESTED** [review]
Thanks for the pull request. It looks like this issue is related to illegal use of `=` in [helm 2.11.0 because it uses Golang 1.10 and does not support this operator](https://github.com/helm/helm/issues/4987#issuecomment-444285485).

The changes I've requested here are to avoid the use of assignments completely, just to remove some complexity here while we're at it. Thanks again!

**Review by philgebhardt — APPROVED** [review]

**Pull request merged**